src/hardware/mixer.cpp

```diff
@@ -2270,12 +2263,12 @@ MixerChannel::~MixerChannel()
 	}
 }
 
-static constexpr int calc_tickadd(const int freq)
+static constexpr float calc_frames_per_tick(const int freq_hz)
```

Comment:
You need fractional numbers if you're supporting _any_ sample rate. Sure, 44100 Hz or 48000 Hz is divisible by 1000, and you get an integer. But 22050 / 1000 = 22.05, and 11025 / 1000 = 11.25.

Currently you can also set any sample rate. `rate = 23456` will just work, try it. It you clamped the frames per tick to an integer here, that would cause issues.

Sure, using a fictional 23456 sample rate is pointless, but it works due to SDL's extra resampling step behind the scenes. Also, in my next PR I'm gonna relax the `rate` setting completely, so you can specify the 49716 OPL rate, or the 32000 MT-32 and SC-55 rate. Runnin the mixer at these weird rates are useful when you want to capture a bit-identical stream without any extra resampling for debugging and other such purposes when you need a WAV with the _exact_ generated sample stream. So I'm gonna allow it (with a warning in the logs).

All in all, it's much preferable to have a mixer that supports _any_ sample rate (within sane limits), and that's how it always has been.

Also, in general, it's best to use floats all the way throughout in DSP code, then convert to int where you really have to at the "boundaries". Internally, you often need to multiply numbers, etc., so it's rather pointless (and inefficient) to store something that can only be an integer as an `int` instead of just uniformly using `floats`. Converting between ints and floats has a performance impact too, and it's inconvenient to cast.

But, again, here we absolutely need fractional numbers.